wallet-new/server/Cardano/Wallet/API/V1/Swagger.hs

```diff
@@ -503,6 +583,404 @@ Disable TLS (Not Recommended)
+  -H 'Accept: application/json;charset=utf-8' \
+  --cacert ./scripts/tls-files/ca.crt
+```
+
+For a broader view, the full list of accounts of a given wallet can be retrieved using [`GET /api/v1/wallets/{{walletId}}/accounts`](#tag/Accounts%2Fpaths%2F~1api~1v1~1wallets~1{walletId}~1accounts%2Fget)
+```
+curl -X GET \
+  https://127.0.0.1:8090/api/v1/wallets/Ae2tdPwUPE...8V3AVTnqGZ/accounts \
+  -H 'Accept: application/json;charset=utf-8' \
+  --cacert ./scripts/tls-files/ca.crt
+```
+
+```json
+{
+    "status": "success",
+    "data": [
+        {
+            "index": 2147483648,
+            "addresses": [
+                "DdzFFzCqrh...U6AGwxWmw"
+            ],
+            "amount": 0,
+            "name": "Initial account",
+            "walletId": "Ae2tdPwUPE...8V3AVTnqGZ"
+        },
+        {
+            "index": 2902829384,
+            "addresses": [
+                "DdzFFzCqrh...Sj5a2rkx8y"
+            ],
+            "amount": 0,
+            "name": "MyOtherAccount",
+            "walletId": "Ae2tdPwUPE...8V3AVTnqGZ"
+        }
+    ],
+    "meta": {
+        "pagination": {
+            "totalPages": 1,
+            "page": 1,
+            "perPage": 10,
+            "totalEntries": 2
+        }
+    }
+}
+```
+
+
+Managing Addresses
+------------------
+
+By default, wallets you create are provided with an account which has one default address. It
+is possible (and recommended) for an account to manage multiple addresses. Address reuse
+actually reduces privacy for it tights more transactions to a small set of addresses.
+
+Note that when paying, this is fairly abstracted from you: addresses are selected from an
+account of your wallet based on different strategies and policies.
+
+To create a new address, use the [`POST /api/v1/addresses`](#tag/Addresses%2Fpaths%2F~1api~1v1~1addresses%2Fpost)
+endpoint:
+
+```
+curl -X POST \
+  https://localhost:8090/api/v1/addresses \
+  -H 'Content-Type: application/json;charset=utf-8' \
+  -H 'Accept: application/json;charset=utf-8' \
+  --cacert ./scripts/tls-files/ca.crt \
+  -d '{
+	"walletId": "Ae2tdPwUPE...V3AVTnqGZ4",
+	"accountIndex": 2147483648
+}'
+```
+
+```json
+{
+    "status": "success",
+    "data": {
+        "id": "DdzFFzCqrh...QfiU5E6N8S",
+        "balance": 0,
+        "used": false,
+        "changeAddress": false
+    },
+    "meta": { ... }
+}
+```
+
+If your wallet is protected with a password, this password is also required in order to create
+new addresses for that wallet. In such case, the field `spendingPassword` should match the one
+defined earlier to protect your wallet.
+
+Addresses generated as just described are always valid. However, when dealing with other
+parties, you might want to check whether a given address is a valid base58-encoded address.
+
+For that purpose, use [`GET /api/v1/addresses/{{address}}/validity`](#tag/Addresses%2Fpaths%2F~1api~1v1~1addresses~1{address}~1validity%2Fget):
```

Comment:
:warning:  We removed the `validity` endpoint. The endpoint is now just `api/v1/addresses/#{Text}`, and it returns a `WalletResponse WalletAddress`. If the address is invalid, it returns a 400 error. :warning:

Reply:
Well spotted @parsonsmatt -- Did miss this one.

Reply:
Interesting! Good catch :)